src/hotspot/share/ci/ciEnv.cpp

```diff
@@ -832,6 +833,14 @@ ciMethod* ciEnv::get_method_by_index_impl(const constantPoolHandle& cpool,
     // As with other two-component call sites, both values must be independently verified.
     assert(index < cpool->cache()->resolved_indy_entries_length(), "impossible");
     Method* adapter = cpool->resolved_indy_entry_at(index)->method();
+    if (is_precompiled()) {
+        ResolvedIndyEntry* indy_info = cpool()->resolved_indy_entry_at(index);
```

Comment:
Can move `indy_info` up and use it in `adapter = ` expression too?

Do you need to use `()` for `cool()`? Other code does not use it.

Reply:
I ment to move up not down:
```
   ResolvedIndyEntry* indy_info = cpool->resolved_indy_entry_at(index);
   Method* adapter = indy_info->method();
#if INCLUDE_CDS
    if (is_precompiled() && !AOTConstantPoolResolver::is_resolution_deterministic(cpool(), indy_info->constant_pool_index())) {
```

Reply:
But `indy_info` is not used outside of that scope, so we shouldn't move it up.

Reply:
What expression at the line #835 uses? Is it different?:
```
  Method* adapter = cpool->resolved_indy_entry_at(index)->method();
```

Reply:
Oops, sorry I misunderstood. I updated the code as you suggested.